### PR TITLE
Makefile: clean up & use targets where possible

### PR DESCRIPTION
**Pull request by unclejack**
This PR cleans up the Makefile a bit. It makes the following changes:
- it uses the existing targets where possible
`make foo && make bar` is the same as `make foo bar`
- `export VARNAME=VARVALUE \ &&` is used instead instead of `VARNAME=VARVALUE command1, VARNAME=VARVALUE command2` to avoid needless repetition 
- the same change is made for targets where we need to change the directory for multiple commands

The changes are kept to a minimum for this PR to make it easier to review and to keep track of the changes.

### CODE REVIEW

**Comment by unclejack** [comment]
The CI status is green. PTAL

**Comment by jojimt** [comment]
@abhinandanpb @rchirakk Please review.

File: Makefile [hunk 1]
```diff
@@ -90,10 +90,7 @@ install-shell-completion:
 	sudo cp scripts/contrib/completion/bash/netctl /etc/bash_completion.d/netctl
 endif
 
-build:
-	make start
-	make ssh-build
-	make stop
+build: start ssh-build stop
```
Comment (rchirakk) [inline Makefile]: is n't this going to break parallel builds ? -jX

Comment (unclejack) [inline Makefile]: @rchirakk -jX shouldn't be used at all in the context of this repository. It doesn't make sense.

Comment (unclejack) [inline Makefile]: We're already using something similar in the following places:
https://github.com/contiv/netplugin/blob/master/Makefile#L33
https://github.com/contiv/netplugin/blob/master/Makefile#L81
https://github.com/contiv/netplugin/blob/master/Makefile#L188
https://github.com/contiv/netplugin/blob/master/Makefile#L227

We would've run into this already. Please let me know if you run into any issues, with or without this PR.

**Comment by dseevr** [comment]
@unclejack this needs a rebase

**Comment by unclejack** [comment]
@dseevr Thanks! I've rebased the PR.

**Comment by unclejack** [comment]
This PR is green after the rebase. PTAL

**Comment by dseevr** [comment]
@abhinandanpb @rchirakk ping

**Comment by rhim** [comment]
@unclejack fix conflicts, then merge

**Comment by unclejack** [comment]
I've fixed the conflicts and the tests pass. I'll merge this now.

**Pull request merged**